### PR TITLE
Fix breaking.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/analysis/ConstantEvaluator.cpp [hunk 1]
```diff
@@ -255,6 +258,8 @@ optional<TypedRational> constantToTypedValue(Type const& _type)
 		return nullopt;
 }
 
+}
```
Comment (bshastry) [inline libsolidity/analysis/ConstantEvaluator.cpp:261]: Please add a comment denoting end of anon namespace for better readability

File: cmake/EthCompilerSettings.cmake [hunk 2]
```diff
@@ -60,7 +60,6 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 	eth_add_cxx_compiler_flag_if_supported(-Wfinal-dtor-non-final-class)
 	eth_add_cxx_compiler_flag_if_supported(-Wnewline-eof)
 	eth_add_cxx_compiler_flag_if_supported(-Wsuggest-destructor-override)
-	eth_add_cxx_compiler_flag_if_supported(-Wunreachable-code-break)
```
Comment (bshastry) [inline cmake/EthCompilerSettings.cmake:63]: This change probably does not belong in this PR?

File: scripts/docs_version_pragma_check.sh [hunk 3]
```diff
@@ -119,8 +119,13 @@ function findMinimalVersion()
...
     then
-        printError "No release $sign$pragmaVersion was listed in available releases!"
-        exit 1
+        if [[ "$greater" = true && "$pragmaVersion" =~ 99 ]]
+        then
+            printError "Skipping version check for pragma: $pragmaVersion"
+        else
+            printError "No release $sign$pragmaVersion was listed in available releases!"
+            exit 1
+        fi
```
Comment (bshastry) [inline scripts/docs_version_pragma_check.sh:128]: This change does not belong here perhaps?

**Review by bshastry — CHANGES_REQUESTED** [review]

File: scripts/docs_version_pragma_check.sh (hunk 3 above)
Comment (chriseth) [reply scripts/docs_version_pragma_check.sh:128]: Yes, the scope of this PR got extended to "fix breaking".

File: libsolidity/analysis/ConstantEvaluator.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/analysis/ConstantEvaluator.cpp:261]: Isn't it obvious? Do we use it in other places?

**Comment by bshastry** [comment]
SMT checker tests failing and base branch out-of-date. PR needs rebase maybe?

File: libsolidity/analysis/ConstantEvaluator.cpp (hunk 1 above)
Comment (axic) [reply libsolidity/analysis/ConstantEvaluator.cpp:261]: > Do we use it in other places?

We do not have comments in other place. We use anonymous namespaces extensively.

**Review by hrkrshnn — APPROVED** [review]

File: scripts/docs_version_pragma_check.sh [hunk 4]
```diff
@@ -119,8 +119,13 @@ function findMinimalVersion()
 
     if [ -z "$version" ]
     then
-        printError "No release $sign$pragmaVersion was listed in available releases!"
-        exit 1
+        if [[ "$greater" = true && "$pragmaVersion" =~ 99 ]]
```
Comment (bshastry) [inline scripts/docs_version_pragma_check.sh:122]: Does this mean that if version string is zero, instead of exiting with `1`, we check if `greater` is true and `pragmaVersion`  equals the regex expression `99`, so something like `>99`? Could you briefly explain why this is needed?

**Comment by hrkrshnn** [comment]
@bshastry The SMTChecker failures are just flaky test generating different examples.

File: libsolidity/analysis/ConstantEvaluator.cpp (hunk 1 above)
Comment (bshastry) [reply libsolidity/analysis/ConstantEvaluator.cpp:261]: I don't know if we use it elsewhere but it took me a second pass to ensure only the said functions were in the anonymous namespace and not others.

File: scripts/docs_version_pragma_check.sh (hunk 4 above)
Comment (chriseth) [reply scripts/docs_version_pragma_check.sh:122]: For code snippets that require 0.8.0, we cannot write `>= 0.8.0` because that version does not exist. Instead, we write `>0.7.99`, which is correct before and after the release.

It is just a quick fix. After the 0.8 release, we should change the version pragmas in the documentation and this fix should maybe also be improved at some point.
